api/events/event_handler_test.go

```diff
@@ -49,13 +47,70 @@ func Test_EventHandler_GetEventsForRadixApplication(t *testing.T) {
 	)
 }
 
-func createKubernetesEvent(client *kubefake.Clientset, ns, name, involvedObjectName string) {
-	client.CoreV1().Events(ns).CreateWithEventNamespace(&v1.Event{
+func Test_EventHandler_GetEvents_PodState(t *testing.T) {
+	appName, envName := "app", "env"
+	appNamespace := k8sObjectUtils.GetEnvironmentNamespace(appName, envName)
+
+	t.Run("ObjectState is nil for normal event type", func(t *testing.T) {
+		kubeClient := kubefake.NewSimpleClientset()
+		createKubernetesEvent(kubeClient, appNamespace, "ev1", "Normal", "pod1", "Pod")
+		createKubernetesPod(kubeClient, "pod1", appNamespace, true, true, 0)
+		ra := builders.NewRadixApplicationBuilder().WithAppName(appName).BuildRA()
+		eventHandler := Init(kubeClient)
+		events, _ := eventHandler.GetEvents(RadixEnvironmentNamespace(ra, envName))
+		assert.Len(t, events, 1)
+		assert.Nil(t, events[0].InvolvedObjectState)
+	})
+
+	t.Run("ObjectState has Pod state for warning event type", func(t *testing.T) {
+		kubeClient := kubefake.NewSimpleClientset()
+		createKubernetesEvent(kubeClient, appNamespace, "ev1", "Warning", "pod1", "Pod")
+		createKubernetesPod(kubeClient, "pod1", appNamespace, true, false, 0)
+		ra := builders.NewRadixApplicationBuilder().WithAppName(appName).BuildRA()
+		eventHandler := Init(kubeClient)
+		events, _ := eventHandler.GetEvents(RadixEnvironmentNamespace(ra, envName))
+		assert.Len(t, events, 1)
+		assert.NotNil(t, events[0].InvolvedObjectState)
+		assert.NotNil(t, events[0].InvolvedObjectState.Pod)
+	})
+
+	t.Run("ObjectState is nil for warning event type when pod not exist", func(t *testing.T) {
+		kubeClient := kubefake.NewSimpleClientset()
+		createKubernetesEvent(kubeClient, appNamespace, "ev1", "Normal", "pod1", "Pod")
+		ra := builders.NewRadixApplicationBuilder().WithAppName(appName).BuildRA()
```

Comment:
Could also include ra in initializer if appname is not variable

Reply:
Agree. Fixed.